### PR TITLE
YOLO26

### PR DESCRIPTION
**Pull request by asabri97**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
Updates the iOS app and YOLO Swift package to prioritize **YOLO26** by default, add **automatic image orientation handling**, and improve **model loading/download + external display reliability** 🚀📱

### 📊 Key Changes
- **Default models switched to YOLO26n** in example apps (SwiftUI + UIKit) for detection 🆕
- **Automatic image orientation normalization** added to the YOLO package:
  - New `normalizeImageOrientation(_:)` utility
  - `YOLO(uiImage)` now normalizes orientation internally (examples removed their custom fixes) 🖼️✅
  - Added/updated unit tests to verify normalization across all `UIImage.Orientation` values 🧪
- **Better YOLO26 output handling** in the core predictors:
  - Stores the model URL to detect “YOLO26” models 🔎
  - Adds score normalization helpers (handles logits / 0–100 style outputs) 🎯
  - Updates **ObjectDetector / Segmenter / PoseEstimator / Classifier** to support YOLO26-style outputs and post-processing (including conditional NMS behavior) ⚙️
- **UI enhancement:** adds a **YOLO version toggle menu (YOLO11 ↔ YOLO26)** in the main app UI 🧭
  - Model selection logic now prefers YOLO26 when available (with smarter picking of local vs remote + version) 📌
- **Model download robustness improvements**:
  - Validates downloaded file size, checks ZIP integrity before extraction, cleans up partial files, and handles download failures more safely 📦🛡️
  - Cancels in-progress downloads when starting a new model load 🧹
- **Stability & cleanup improvements**:
  - Better error handling when model URL is missing (no more `fatalError`) 🧯
  - Adds `deinit` cleanup to stop camera capture / remove observers (YOLOView, VideoCapture, SceneDelegate) 🧼
  - External display detection improved for iOS 16+ scene-based setups, and UI updates moved to `@MainActor` for safety 🖥️

### 🎯 Purpose & Impact
- **Easier “it just works” image inference**: users no longer need to manually fix photo orientation—fewer wrong-rotation predictions 📸✅
- **Better YOLO26 support across tasks** (detect/seg/pose/cls): makes YOLO26 models behave correctly even when their output format differs from YOLO11 🧠⚡
- **Improved app reliability**:
  - Fewer crashes during model load, safer camera/session lifecycle handling, and more predictable external-display behavior 🔒📺
- **Smoother model workflows**:
  - More robust downloads + extraction reduces “bad zip / partial download” issues 🧳
  - Version toggle lets users quickly compare YOLO26 vs YOLO11 performance/compatibility 🎛️

### CODE REVIEW

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Solid overall update: defaulting example apps to YOLO26, adding YOLO26-aware post-processing paths, and moving image-orientation normalization into the package are all good improvements.

A few correctness issues need attention before merge: (1) external display detection uses `openSessions`/`scene` in a way that likely won’t compile, and (2) `YOLO.callAsFunction(UIImage)` still force-unwraps `CIImage(image:)`, which can crash on valid UIImage variants. Also, the YOLO26 pose/segmentation post-processing should guard against a `(0,0)` `modelInputSize` to avoid NaN/invalid results. Performance-wise, per-detection `MLMultiArray` allocations in YOLO26 segmentation could be costly in realtime scenarios.

💬 Posted 5 inline comments

File: Sources/YOLO/Segmenter.swift [hunk 1]
```diff
@@ -368,6 +406,84 @@ public class Segmenter: BasePredictor, @unchecked Sendable {
...
+      let clampedX = max(0.0, min(CGFloat(modelWidth), boxX))
+      let clampedY = max(0.0, min(CGFloat(modelHeight), boxY))
+      let clampedW = max(0.0, min(CGFloat(modelWidth) - clampedX, boxW))
+      let clampedH = max(0.0, min(CGFloat(modelHeight) - clampedY, boxH))
+
+      let box = CGRect(x: clampedX, y: clampedY, width: clampedW, height: clampedH)
+
+      // Extract mask coefficients (32 values starting at offset + 6)
+      guard
+        let maskProbs = try? MLMultiArray(
```
Comment (UltralyticsAssistant) [inline Sources/YOLO/Segmenter.swift]: 💡 **MEDIUM**: This allocates a new `MLMultiArray` for mask coefficients for every detection (up to `numDetections`, typically ~300). That’s potentially expensive per frame. If this path is used for realtime, consider extracting mask coeffs into a reusable buffer/array and only creating an `MLMultiArray` for selected detections after filtering (confidence/NMS).
